### PR TITLE
Update non-weight-based fees

### LINKED ISSUE #30: Update non-weight-based fees
[This section](https://github.com/substrate-developer-hub/knowledgebase/blob/master/current/runtime/fees.md#additional-fees) is out of date as all fees are part of the Transaction Payment pallet.

The [Polkadot Wiki](https://wiki.polkadot.network/docs/en/learn-transaction-fees#other-resource-limitation-strategies) is more accurate, although this will need to be a bit more generic for Substrate.

### PR DESCRIPTION
**Pull request by danforbes**
Closes #30

### CODE REVIEW

File: current/runtime/fees.md [hunk 1]
```diff
@@ -89,16 +89,20 @@ transactions than the target, fees are decreased by a small amount. More informa
...
+
+- Bonds: Some transactions, like voting, may require a bond that will be returned or slashed after
+  an on-chain event. In the voting example, returned at the end of the election or slashed if the
+  voter tried anything malicious.
+- Deposits: Some transactions use storage space indefinitely and may require a deposit that will be
+  returned if the user decides to free storage.
+- Burns: A transaction may burn funds internally based on its logic. For example, a transaction may
+  burn funds from the sender if it creates new storage entries, thus increasing the state size.
+- Limits: Some limits are part of the protocol. For example, in Polkadot, nominators can only nominate
+  16 validators. This limits the complexity of the validator election process.
```
Comment (joepetrowski) [inline current/runtime/fees.md]:
```suggestion
- Limits: You can code arbitrary limits into a pallet. For example, the default Staking pallet only
  allows nominators to nominate up to 16 validators to limit the complexity of the validator
  election process.
```

File: current/runtime/fees.md [hunk 2]
```diff
@@ -89,16 +89,20 @@ transactions than the target, fees are decreased by a small amount. More informa
...
+## Additional Fees
+
+Inclusion fees must be computable prior to execution, and therefore can only represent fixed logic.
+Some transactions warrant limiting resources with other strategies. For example:
+
+- Bonds: Some transactions, like voting, may require a bond that will be returned or slashed after
+  an on-chain event. In the voting example, returned at the end of the election or slashed if the
+  voter tried anything malicious.
+- Deposits: Some transactions use storage space indefinitely and may require a deposit that will be
+  returned if the user decides to free storage.
```
Comment (joepetrowski) [inline current/runtime/fees.md]:
```suggestion
- Deposits: Some transactions use storage space indefinitely. You may want to require a deposit
  that will be returned if the user decides to free storage.
```

File: current/runtime/fees.md [hunk 3]
```diff
@@ -89,16 +89,20 @@ transactions than the target, fees are decreased by a small amount. More informa
...
-types of additional fees, such as bonds, which may be returned or slashed based on some future
-outcome or deposits, which may be returned after some future event (like freeing up storage).
+## Additional Fees
+
+Inclusion fees must be computable prior to execution, and therefore can only represent fixed logic.
+Some transactions warrant limiting resources with other strategies. For example:
+
+- Bonds: Some transactions, like voting, may require a bond that will be returned or slashed after
+  an on-chain event. In the voting example, returned at the end of the election or slashed if the
+  voter tried anything malicious.
```
Comment (joepetrowski) [inline current/runtime/fees.md]:
```suggestion
- Bonds: Some transactions may warrant a bond that will be returned or slashed after an on-chain
  event. For example, you may want to take a bond to vote in a referendum that will be returned at
  the end of the election or slashed if the voter tries anything malicious.
```

**Review by joepetrowski — COMMENTED** [review]
Better, just needs to be a bit more generic since we're talking about custom chains and not a single one (Polkadot). I.e.

- Polkadot: The protocol requires X because...
- Substrate: You may want to do X because...

**Comment by danforbes** [comment]
@joepetrowski thanks for those suggestions. I took the gist of what I interpreted you to be saying and put my own spin on things. Let me know if I went overboard and we can revert to your original suggestions or some combination of the two :+1:

**Review by joepetrowski — APPROVED** [review]

**Pull request merged**